packages/plugin-flex/src/commands/flex/plugins/archive/plugin-version.ts

```diff
@@ -93,8 +93,13 @@ export default class FlexPluginsArchivePluginVersion extends ArchiveResource<Plu
       runtime: 'node12',
     };
 
-    // TODO: remove Environment when the other ticket is done
     if (request.assetVersions?.length === 0 && request.functionVersions?.length === 0) {
+      const environment = await this.serverlessClient.getEnvironment(build.sid, this._flags.name);
+      if (!environment.sid) {
```

Comment:
You should return null, then `if (!environment) { reutrn; }`

Reply:
Also, you shouldn't `return` - there are other things that are happening after this `if` statement, instead it should be:

```ts
const environment = await this.serverlessClient.getEnvironment(build.sid, this._flags.name);
if (environment) {
  await this.serverlessClient.deleteEnvironment(build.sid, environment.sid);
}
```

Reply:
hmm, but originally we were doing `return;` if it reached this block?

Reply:
i'll update it to what you suggested bc it's cleaner, but it seems like i should be leaving the last return in